docs/source/user-guide/dataset_object.rst

```diff
@@ -83,46 +81,57 @@ whether the column is categorical or not:
 * else:
     *  number of unique < `max_categories` AND (number of unique / number of samples) < `max_categorical_ratio`
 
-Creating a Dataset
-======================
+Useful Functions
+================
+
+Create Dataset From a Numpy Arrays
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A Dataset can be created using a 2D numpy array for features and 1D numpy array for the labels. The features array is mandatory, and the labels array is optional.
+
+    >>> features = np.array([[0.25, 0.3, 0.3], [0.14, 0.75, 0.3], [0.23, 0.39, 0.1]])
+    >>> labels = np.array([0.1, 0.1, 0.7])
+    >>> dataset_with_labels = Dataset.from_numpy(features, labels)
+    >>> dataset_without_labels = Dataset.from_numpy(features)
+
+Also, it's possible to assign names to the features and label:
+
+    >>> Dataset.from_numpy(
+    ...     features, labels,
+    ...     feature_names=['feat1', 'feat2', 'feat3',],
+    ...     label_name='target'
+    ... )
+
+All the rest of the Dataset's properties can be passed also as a regular keyword arguments:
+
+    >>> Dataset.from_numpy(
+    ...     features, labels,
+    ...     feature_names=['feat1', 'feat2', 'feat3',],
+    ...     label_name='target',
+    ...     max_float_categories=10
+    ... )
 
-From a DataFrame
+
+Train Test Split
 ~~~~~~~~~~~~~~~~
-The default dataset constructor is expecting to get a dataframe. the rest of the properties
-are optional, but if your data have date/index/label you would want to define them.
 
-.. code-block:: python
+Using the `sklearn.model_selection.train_test_split <https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html>`_ but also copies the metadata to each instance of the split.
 
-    Dataset(my_dataframe, features=['feat1', 'feat2', 'feat3'], label='target', index='id', date='timestamp')
+    >>> train_dataset, test_dataset = dataset.train_test_split()
 
-From Numpy Arrays
-~~~~~~~~~~~~~~~~~~~
-A Dataset can be created using a 2D numpy array for features and 1D numpy array for the labels. The features array is mandatory, and the labels array is optional.
 
-.. code-block:: python
+Copy
+~~~~
 
-    features = np.array([[0.25, 0.3, 0.3], [0.14, 0.75, 0.3], [0.23, 0.39, 0.1]])
-    labels = np.array([0.1, 0.1, 0.7])
-    dataset_with_labels = Dataset.from_numpy(features, labels)
-    dataset_without_labels = Dataset.from_numpy(features)
+Copy allows copying of metadata on a new dataframe. This can be used on data that already is passed the train test split.
```

Comment:
rephrase
(second sentence is not clear, but in general this line can be improved)

Reply:
also maybe give a different example, to not confuse the user the the train_test_split which is right above
(they might think "wait, after I do the train_test_split I need to copy the metadata?"